centreon/www/class/centreonLDAP.class.php

```diff
@@ -316,19 +316,42 @@ public function replaceFilter($name): string
         return $name;
     }
 
+    /**
+     * Escape special characters in LDAP filter value for username and group
+     *
+     * @param string $name The attribute to escape
+     * @return string The escaped string
+     */
+    public function escapeLdapFilterSpecialChars($name): string
+    {
+        $specialChars = [
+            '\\' => '\\5C',
+            '*' => '\\2A',
+            '(' => '\\28',
+            ')' => '\\29',
+            '\0' => '\\00',
+        ];
```

Comment:
If those chars comes from a RFC could. you add the link to the RFC into the PHP Doc ?